internal/types/array.go

```diff
@@ -55,6 +60,22 @@ func MustNewArray(values ...any) *Array {
 
 func (a *Array) compositeType() {}
 
+func (a *Array) String() string {
+	b := new(strings.Builder)
+	b.WriteString("[")
+	for i := 0; i < a.Len(); i++ {
+		// TODO obviously this won't work correctly with structs
+		b.WriteString(fmt.Sprintf(" %v", must.NotFail(a.Get(i))))
```

Comment:
No, it is better to create a recursive function `.String()` for all types, including a document.

Reply:
I agree! 
But if I understand correctly what you are suggesting, are you sure it should be in this PR? I've addressed this issue in a comment: https://github.com/FerretDB/FerretDB/pull/518#issuecomment-1114250719

I'm just saying that we can't call `.String()` recursively because of the use of in-built types

I would suggest leaving it as is and later refactoring it, WDYT?

Reply:
I guess we first should decide what this function _does_, as there is no documentation comment. It returns some string representation of array… for what? If that's for debugging, then can we use `testutil.Dump` or some helper like `testutil.AssertEqual` that uses it instead? If that's for not-testing code, then what makes this string representation more important than others, given that this is a method that can be only one?

Reply:
There are situations where Mongo returns an error, that contains user input, for example:
```
values> db.values.find({name:"array-embedded"}, {value:{$slice:[1,2,3,4]}})
MongoServerError: Invalid $slice syntax. The given syntax { $slice: [ 1, 2, 3, 4 ] } did not match the find() syntax because :: Location31272: $slice array argument should be of form [skip, limit] :: The given syntax did not match the expression $slice syntax. :: caused by :: Expression $slice takes at least 2 arguments, and at most 3, but 4 were passed in.
```

In this PR, I emulated this particular error with the following snippet:

```
	if arr.Len() < 2 || arr.Len() > 3 {
			return nil, NewErrorMsg(ErrInvalidArg,
				fmt.Sprintf(
					"Invalid $slice syntax. The given syntax { $slice: %s } "+
						"did not match the find() syntax because :: Location31272: "+
						"$slice array argument should be of form [skip, limit] :: "+
						"The given syntax did not match the expression "+
						"$slice syntax. :: caused by :: "+
						"Expression $slice takes at least 2 arguments, and at most 3, but %d were passed in.",
					projectionValue,
					arr.Len(),
				))
		}
```

So, attaching `.String()` to FerretDB types would make it convenient to print these values when returning an error message to the user

Reply:
Well, the end-user does not care whatever this code is in method or function, or how it is named :)

The problem with this code being in the `String` method is that `fmt.Stringer` is being used implicitly in many cases, and that might be problematic. For example, that might be confusing in tests where simple assert.Equal uses `fmt.Sprint`. So let's not introduce a method with that name.

So let's use a name that does not match any common interface. `JSONSyntax` might be a good name. And let's add a documentation comment.